core/src/test/java/io/questdb/test/griffin/CreateMatViewTest.java

```diff
@@ -0,0 +1,264 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin;
+
+import io.questdb.griffin.SqlException;
+import io.questdb.test.AbstractCairoTest;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Test;
+
+import static org.junit.Assert.fail;
+
+public class CreateMatViewTest extends AbstractCairoTest {
+    private static final String TABLE1 = "table1";
+    private static final String TABLE2 = "table2";
+    private static final String TABLE3 = "table3";
+
+    @Test
+    public void testCreateMatViewGroupByTimestampTest() throws Exception {
+        assertMemoryLeak(() -> {
+            createTable(TABLE1);
+
+            ddl("create materialized view test as select timestamp_floor('30s', ts) as ts, avg(v) from " + TABLE1 + " order by ts");
+
+            final String expected = "ts\tavg\n" +
+                    "1970-01-01T00:00:00.000000Z\t1.0\n" +
+                    "1970-01-01T00:00:30.000000Z\t4.0\n" +
+                    "1970-01-01T00:01:00.000000Z\t7.0\n";
+
+            assertQuery(
+                    expected,
+                    "test",
+                    "ts",
+                    true,
+                    true
+            );
+        });
+    }
+
+    @Test
+    public void testCreateMatViewInvalidSampleByTest() throws Exception {
+        assertMemoryLeak(() -> {
+            createTable(TABLE1);
+
+            try {
+                ddl("create materialized view test as select ts, avg(v) from " + TABLE1 + " sample by 3M");
+                fail("Expected SqlException missing");
+            } catch (SqlException e) {
+                TestUtils.assertContains(e.getFlyweightMessage(), "Materialized view query with invalid sampling interval, 'M' and 'y' intervals are not supported");
```

Comment:
I'd rather support all the intervals

Reply:
This is a tricky one because `M` and `y` intervals cannot be converted to micros.
As discussed, I will change the view definition model, and instead of micros it will store the interval + unit.
I will also add sample by from, to and timezone fields.

Reply:
@ideoma, pushed the refactored MaterializedViewDefinition yesterday, it takes the sampling period and unit now.
Will be working on setting sample by from, to, timezone and tz offset now.

Reply:
done